aesara/link/numba/dispatch.py

```diff
@@ -1334,3 +1410,400 @@ def broadcastto(x, *shape):
         return ret
 
     return broadcastto
+
+
+def int_to_float_fn(inputs, out_dtype):
+    """Create a Numba function that converts integer and boolean ``ndarray``s to floats."""
+
+    if any(i.type.numpy_dtype.kind in "ib" for i in inputs):
+
+        args_dtype = np.dtype(f"f{out_dtype.itemsize}")
+
+        @numba.njit(inline="always")
+        def inputs_cast(x):
+            return x.astype(args_dtype)
+
+    else:
+
+        @numba.njit(inline="always")
+        def inputs_cast(x):
+            return x
+
+    return inputs_cast
+
+
+@numba_funcify.register(Dot)
+def numba_funcify_Dot(op, node, **kwargs):
+    # Numba's `np.dot` does not support integer dtypes, so we need to cast to
+    # float.
+
+    out_dtype = node.outputs[0].type.numpy_dtype
+    inputs_cast = int_to_float_fn(node.inputs, out_dtype)
+
+    @numba.njit
+    def dot(x, y):
+        return np.dot(inputs_cast(x), inputs_cast(y)).astype(out_dtype)
+
+    return dot
+
+
+@numba_funcify.register(Softmax)
+def numba_funcify_Softmax(op, node, **kwargs):
+
+    x_at = node.inputs[0]
+    x_dtype = x_at.type.numpy_dtype
+    x_dtype = numba.np.numpy_support.from_dtype(x_dtype)
+
+    # np.max(x, axis=1)
+    reduce_max = create_axis_reducer(np.maximum, -np.inf, 1, x_at.ndim, x_dtype)
+    # np.sum(x, axis=1)
+    reduce_sum = create_axis_reducer(np.add, 0.0, 1, x_at.ndim, x_dtype)
+
+    @numba.njit
+    def softmax(x):
+        z = np.expand_dims(reduce_max(x), -1)
+        e_x = np.exp(x - z)
+        w = np.expand_dims(reduce_sum(e_x), -1)
+        sm = e_x / w
+        return sm
+
+    return softmax
+
+
+@numba_funcify.register(LogSoftmax)
+def numba_funcify_LogSoftmax(op, node, **kwargs):
+
+    x_at = node.inputs[0]
+    x_dtype = x_at.type.numpy_dtype
+    x_dtype = numba.np.numpy_support.from_dtype(x_dtype)
+
+    # np.max(x, axis=1)
+    reduce_max = create_axis_reducer(np.maximum, -np.inf, 1, x_at.ndim, x_dtype)
+    # np.sum(x, axis=1, keepdims=True)
+    reduce_sum = create_axis_reducer(np.add, 0.0, 1, x_at.ndim, x_dtype, keepdims=True)
+
+    @numba.njit
+    def log_softmax(x):
+        xdev = x - np.expand_dims(reduce_max(x), -1)
+        lsm = xdev - np.log(reduce_sum(np.exp(xdev)))
+        return lsm
+
+    return log_softmax
+
+
+@numba_funcify.register(ScalarSoftplus)
+def numba_funcify_ScalarSoftplus(op, node, **kwargs):
+
+    not_int8 = node.inputs[0].dtype not in ("int8", "uint8")
+
+    @numba.njit
+    def scalarsoftplus(x):
+        if x < -37.0:
+            return np.exp(x) if not_int8 else np.exp(x)
+        elif x < 18.0:
+            return np.log1p(np.exp(x)) if not_int8 else np.log1p(np.exp(x))
```

Comment:
I don't see a difference between the two.

Reply:
That code is copied from the underlying `Op`.  The `else` clauses were supposed to convert the underlying dtypes, but the argument isn't supported in Numba, so it had to be removed.

I've updated this so that it casts to the output dtype.